### PR TITLE
Various test infra updates from testing Azure ops with MAUI test app

### PR DESCRIPTION
**Pull request by skottmckay**
### Description

- fix issue with handling string input
- set minSdkVersion
  - otherwise defaults to 19 which we don't support and the build breaks
- comment out the debug logging hook
  - enabling it breaks the Android native logging
  - can be enabled if you need to debug C# code
- update test data tools to allow creating input data for raw file contents (e.g. audio) and from strings (e.g. auth token value)
- fix some warnings

### Motivation and Context

Improve test setup

### CODE REVIEW

File: csharp/tools/MauiModelTester/Utils.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ internal static OrtValue CreateOrtValueFromTensorProto(Onnx.TensorProto tensorPr
             OrtValue ortValue = null;
 
             // special case for strings
-            if (tensorElementType == typeof(string))
+            if ((TensorElementType)tensorProto.DataType == TensorElementType.String)
```
Comment (edgchen1) [inline csharp/tools/MauiModelTester/Utils.cs:64]: why does `GetElementType(TensorElementType.String)` return `typeof(byte)` and not `typeof(string)`?

File: tools/python/onnx_test_data_utils.py [hunk 2]
```diff
@@ -210,6 +220,21 @@ def get_arg_parser():
 
         data = create_random_data(args.shape, args.datatype, args.min_value, args.max_value, args.seed)
         numpy_to_pb(args.name, data, args.output)
+    elif args.action == "raw_to_pb":
+        if not args.output or not args.name:
```
Comment (edgchen1) [inline tools/python/onnx_test_data_utils.py:224]: we also need `args.input` right?

File: tools/python/onnx_test_data_utils.py [hunk 3]
```diff
@@ -210,6 +220,21 @@ def get_arg_parser():
...
+    elif args.action == "string_to_pb":
+        if not args.output or not args.name:
+            print("Missing argument. Need output and name to be specified.", file=sys.stderr)
+            sys.exit(-1)
+
+        data = np.ndarray((1), dtype=object)
+        data[0] = args.input
+        numpy_to_pb(args.name, data, args.output)
     elif args.action == "update_name_in_pb":
         if not args.input or not args.name:
```
Comment (edgchen1) [inline tools/python/onnx_test_data_utils.py:239]: not related to this PR, but it looks like this should also check for `args.output`

File: csharp/tools/MauiModelTester/Utils.cs (hunk 1 above)
Comment (skottmckay) [reply csharp/tools/MauiModelTester/Utils.cs:64]: Hmmm. Good point. Not sure why it does. Will change it to return string.

File: tools/python/onnx_test_data_utils.py (hunk 3 above)
Comment (skottmckay) [reply tools/python/onnx_test_data_utils.py:239]: that one writes to input file if output isn't specified. the description of the helper mentions this.

Comment (edgchen1) [reply tools/python/onnx_test_data_utils.py:239]: ah, I see. never mind that then :)

**Pull request merged**